src/Resources/app/storefront/src/configuration/adyen.js

```diff
@@ -25,9 +25,9 @@ export default {
     paymentMethodTypeHandlers: {
         'scheme': 'handler_adyen_cardspaymentmethodhandler',
         'ideal': 'handler_adyen_idealpaymentmethodhandler',
-        'klarna': 'handler_adyen_klarnaaccountpaymentmethodhandler',
-        'klarna_account': 'handler_adyen_klarnapaynowpaymentmethodhandler',
-        'klarna_paynow': 'handler_adyen_klarnapaylaterpaynowpaymentmethodhandler',
+        'klarna': 'handler_adyen_klarnapaylaterpaymentmethodhandler',
+        'klarna_account': 'handler_adyen_klarnaaccountpaymentmethodhandler',
+        'klarna_paynow': 'handler_adyen_klarnapaynowpaymentmethodhandler',
```

Comment:
Is this required for `state.data`?

Reply:
Not specifically for `state.data`. It is needed to validate the payment methods that require input before submitting the order here https://github.com/Adyen/adyen-shopware6/blob/develop/src/Resources/app/storefront/src/checkout/confirm-order.plugin.js#L37

Reply:
I mean this change. This does not seem strictly related to the pull request

Reply:
Correct, this is a bugfix necessary for the storedPaymentMethods state.data check, the PR description has both explanations. Do you think we should move it to a separate PR?